### ISSUE TITLE
Location puck is drawn below LineLayer

### ISSUE THREAD

**Issue by basvdijk** [mapbox/mapbox-maps-flutter#517]
When I use:

```
    mapboxMap?.location.updateSettings(
      LocationComponentSettings(
        pulsingEnabled: true,
        showAccuracyRing: true,
        puckBearing: PuckBearing.COURSE,
        puckBearingEnabled: true,
      ),
    );
```

And add a LineLayer:

```
    await mapboxMap?.style.addLayer(
      LineLayer(
        id: 'line_layer',
        sourceId: 'line',
        lineColor: Colors.pinkAccent.value,
        lineTrimOffset: [0.0, 1.0],
        lineWidth: 5.0,
      ),
    );
```

The result is that the puck is drawn below the line. I think the default should be that the puck is always on top. Is this a bug or am I missing a setting?
![image](https://github.com/mapbox/mapbox-maps-flutter/assets/644550/014a6fd4-a25d-4971-ac56-5ede1ff92156)

**Comment by evil159** [comment]
Hi @basvdijk, I'm able to reproduce this on Android, on iOS the location indicator is on top. Filed a bug in the internal issue tracker - https://mapbox.atlassian.net/browse/MAPSAND-1587

**Comment by basvdijk** [comment]
@evil159 The screenshot was taken from iOS. Changing the order of first drawing the line and then turning on the puck sovled it.

**Comment by evil159** [comment]
Interesting, I can't seem to reproduce this on iOS, which version to you use?

**Comment by basvdijk** [comment]
@evil159 iOS 17.4.1 but it might also be related to a race condition with async operations.

**Comment by evil159** [comment]
Ah, sorry, I mean what version of the Mapbox Maps Flutter Plugin do you use?

**Comment by basvdijk** [comment]
Commit https://github.com/mapbox/mapbox-maps-flutter/commit/1daef17c48fd927db442a5d19f7aca06a1ffdc63

**Comment by basvdijk** [comment]
@evil159 I have the same issue for point annotations. The yellow dot is below the line. 
![image](https://github.com/mapbox/mapbox-maps-flutter/assets/644550/dee63c12-0abd-4ec9-8542-93fe107cc618)

I saw `createPointAnnotationManager` has a `below` constructor property but not an `above`. Any thoughts?

**Comment by hussenIbrahim** [comment]
Hello

@basvdijk  did you found any solution or workaround?

**Comment by basvdijk** [comment]
@hussenIbrahim yes, the only solution I could find is to change the order things are drawn. So first the route, then the location puck.

**Comment by SuHlaHlaPhyu** [comment]
draw location puck after creating your polyineAnnotationManager.  this works for me.
`
    bookingPolylineAnnotationManager = await mapBoxController!.annotations
        .createPolylineAnnotationManager(id: "tripPolyline");
    
    mapBoxController!.location
        .updateSettings(mapbox.LocationComponentSettings());
`
